aidial_adapter_vertexai/chat_completion.py

```diff
@@ -86,3 +108,65 @@ async def generate_response(usage: TokenUsage, choice_idx: int) -> None:
 
         if params.max_prompt_tokens is not None:
             response.set_discarded_messages(discarded_messages)
+
+    @override
+    async def tokenize(self, request: TokenizeRequest) -> TokenizeResponse:
+        model = await self.get_model(request)
+
+        outputs: List[TokenizeOutput] = []
+        for input in request.inputs:
+            match input:
+                case TokenizeInputRequest():
+                    outputs.append(
+                        await self.tokenize_request(model, input.value)
+                    )
+                case TokenizeInputString():
+                    outputs.append(
+                        await self.tokenize_string(model, input.value)
+                    )
+                case _:
+                    assert_never(input.type)
+        return TokenizeResponse(outputs=outputs)
+
+    async def tokenize_string(
+        self, model: ChatCompletionAdapter, value: str
+    ) -> TokenizeOutput:
+        try:
+            tokens = await model.count_completion_tokens(value)
+            return TokenizeSuccess(token_count=tokens)
+        except Exception as e:
+            return TokenizeError(error=str(e))
+
+    async def tokenize_request(
+        self, model: ChatCompletionAdapter, request: ChatCompletionRequest
+    ) -> TokenizeOutput:
+        try:
+            prompt = await model.parse_prompt(request.messages)
```

Comment:
May it return an error object instead of a prompt?

Reply:
Indeed. Fixed.